### PR TITLE
Add reviseMesh tool and modify editMaterialID tool

### PR DESCRIPTION
**Pull request by norihiro-w**
This PR introduces
- a new tool reviseMesh based on MeshRevision class
- supporting element type to modify material IDs

### CODE REVIEW

File: Applications/Utils/MeshEdit/reviseMesh.cpp [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+		// write into a file
+		if (new_mesh) {
+			INFO("Revised mesh: %d nodes, %d elements.", new_mesh->getNNodes(), new_mesh->getNElements());
+			FileIO::writeMeshToFile(*new_mesh, output_arg.getValue());
+		}
+
+		delete org_mesh;
+		delete new_mesh;
+	} catch (TCLAP::ArgException &e) {
```
Comment (endJunction) [inline Applications/Utils/MeshEdit/reviseMesh.cpp]: Can the `catch` go to line 52 after last tclap's parse?
Maybe also the `try` part too? Just enclose the `cmd.parse()`?

**Comment by endJunction** [comment]
Looks clean. :fast_forward:

File: Applications/Utils/MeshEdit/editMaterialID.cpp [hunk 2]
```diff
@@ -34,7 +35,12 @@ int main (int argc, char* argv[])
 	TCLAP::CmdLine cmd("Edit material IDs of mesh elements.", ' ', "0.1");
 	TCLAP::SwitchArg replaceArg("r", "replace", "replace material IDs", false);
 	TCLAP::SwitchArg condenseArg("c", "condense", "condense material IDs", false);
-	cmd.xorAdd(replaceArg, condenseArg);
+	TCLAP::SwitchArg specifyArg("s", "specify", "specify material IDs", false);
```
Comment (TomFischer) [inline Applications/Utils/MeshEdit/editMaterialID.cpp]: What exactly can I do with this option? Can you add a little bit more explanation, please?

File: FileIO/writeMeshToFile.cpp [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+#include "FileIO/VtkIO/VtuInterface.h"
+
+namespace FileIO
+{
+void writeMeshToFile(const MeshLib::Mesh &mesh, const std::string &file_name)
+{
+	if (BaseLib::hasFileExtension("msh", file_name))
+	{
+		Legacy::MeshIO meshIO;
+		meshIO.writeToFile(file_name);
```
Comment (TomFischer) [inline FileIO/writeMeshToFile.cpp]: I wonder how the ```meshIO``` object get the mesh data. Did you forget``` meshIO.set(...)```?

File: MeshLib/MeshEditing/ElementValueModification.cpp [hunk 4]
```diff
@@ -91,4 +91,18 @@ unsigned ElementValueModification::condense(MeshLib::Mesh &mesh)
 	return nValues;
 }
 
+unsigned ElementValueModification::set(MeshLib::Mesh &mesh, MeshElemType ele_type, unsigned new_value)
+{
+	std::vector<MeshLib::Element*> elements (mesh.getElements());
```
Comment (TomFischer) [inline MeshLib/MeshEditing/ElementValueModification.cpp]: Why not ```std::vector<MeshLib::Element*> const& elements (mesh.getElements());```? Works with gcc 4.9.2 and clan 3.5.0.

File: FileIO/writeMeshToFile.cpp (hunk 3 above)
Comment (norihiro-w) [inline FileIO/writeMeshToFile.cpp]: sorry I forgot to push one more commit to fix this bug.

File: MeshLib/MeshEditing/ElementValueModification.cpp (hunk 4 above)
Comment (norihiro-w) [inline MeshLib/MeshEditing/ElementValueModification.cpp]: right, i should use reference here.

File: MeshLib/MeshEditing/ElementValueModification.h [hunk 5]
```diff
@@ -35,6 +36,9 @@ class ElementValueModification
 	/// Returns true if successful or false if the value is already taken.
 	static bool replace(MeshLib::Mesh &mesh, unsigned old_value, unsigned new_value, bool replace_if_exists = false);
 
+	/// Sets new value for all elements having the given element type
+	static unsigned set(MeshLib::Mesh &mesh, MeshElemType ele_type, unsigned new_value);
```
Comment (rinkk) [inline MeshLib/MeshEditing/ElementValueModification.h]: you could call this something like ```setByElementType()``` to make more clear what happens here.

File: Applications/Utils/MeshEdit/reviseMesh.cpp (hunk 1 above)
Comment (norihiro-w) [inline Applications/Utils/MeshEdit/reviseMesh.cpp]: i deleted try-catch as it is not so important

**Comment by norihiro-w** [comment]
updated with additional changes in MeshEnums to display a list of element type names

**Pull request merged**